backend/src/aggregations.js

```diff
@@ -0,0 +1,220 @@
+const BN = require("bn.js");
+const nearApi = require("near-api-js");
+
+const { getAllLockupAccountIds } = require("./db-utils");
+const { nearRpc } = require("./near");
+
+let CIRCULATING_SUPPLY = {
+  block_height: undefined,
+  circulating_supply_in_yoctonear: undefined,
+};
+
+// utils from https://github.com/near/account-lookup/blob/master/script.js
+const readOption = (reader, f) => {
+  let x = reader.read_u8();
+  if (x == 1) {
+    return f();
+  }
+  return null;
+};
+
+// viewLockupState function taken from https://github.com/near/account-lookup/blob/master/script.js
+const viewLockupState = async (contractId, blockHeight) => {
```

Comment:
Just something that I noticed and was curious about; is it safe for `viewLockupState()` to retry _indefinitely_ when it encounters _any_ type of error, with no boundary to how many times it retries, and no error reporting if it has to retry many times to finish? Feels like this could permanently deadlock if fed e.g. a bad contractId?  Same with `while(true)` loops on L126 & L152  of this file.

Reply:
@MaximusHaximus Thanks for pointing those out! This is a draft PR that we needed to collect the figures in the first place. Since then we have done more exploration of the space, and @telezhnaya will take it over from here solidifying the implementation and having graceful retry strategies.